### PR TITLE
Fixes #27121 - migrate proxy usage to global content proxy

### PR DESCRIPTION
**Pull request by jlsherrill**
This migrates the following to use the globally set
contetn proxy:

* Repository Discovery
* Manifest actions (refresh, entitlement change)
* Redhat Repository enablement

### CODE REVIEW

File: app/models/katello/concerns/http_proxy_extensions.rb [hunk 1]
```diff
@@ -5,6 +5,12 @@ module HttpProxyExtensions
 
       included do
         after_save :update_default_proxy_setting
+
+        def self.default_global_content_proxy
```
Comment (jjeffers) [inline app/models/katello/concerns/http_proxy_extensions.rb]: Oh, this is a great addition!

Comment (jjeffers) [reply app/models/katello/concerns/http_proxy_extensions.rb]: @jlsherrill should we refactor other queries where we try to find the global default to use this method?

Comment (jlsherrill) [reply app/models/katello/concerns/http_proxy_extensions.rb]: good idea! i'll see if i can find em all

Comment (jjeffers) [reply app/models/katello/concerns/http_proxy_extensions.rb]: I think there will be a a couple more in https://github.com/Katello/katello/pull/8178.

Comment (jlsherrill) [reply app/models/katello/concerns/http_proxy_extensions.rb]: i don't actually see any in the current code base, but will change in once i rebase against #8178

Comment (jlsherrill) [reply app/models/katello/concerns/http_proxy_extensions.rb]: updated, let me know if i missed any

**Pull request merged**